packages/js/src/plugins/auctionHouseModule/createAuctionHouse.ts

```diff
@@ -36,13 +41,15 @@ export type CreateAuctionHouseInput = {
   sellerFeeBasisPoints: number;
   requiresSignOff?: boolean;
   canChangeSalePrice?: boolean;
+  auctioneerScopes?: AuthorityScope[];
 
   // Accounts.
   treasuryMint?: PublicKey;
   payer?: Signer;
   authority?: PublicKey;
   feeWithdrawalDestination?: PublicKey;
   treasuryWithdrawalDestinationOwner?: PublicKey;
+  auctioneerAuthority?: Signer;
```

Comment:
There is an issue here.

The **Auction House authority** does not need to be a signer when creating the Auction House (only the payer does) **BUT** it does need to be a signer when delegating the Auction House to an Auctioneer instance.

Therefore, the `authority` parameter now needs to be of type `PublicKey | Signer`. That means, if someone gives an `authority` as a `PublicKey` AND provides an `auctioneerAuthority`, we should throw a custom error explaining what happened.

Also, `auctioneerAuthority` does not need to be a signer here.

Note that this works in the tests because of the default values. Could you please create a new test for every operation you implement that ensures ALL public keys and/or signers are provided as separate wallets so these issues can surface?

```suggestion
  payer?: Signer;
  authority?: PublicKey | Signer; // Authority is required to sign when delegating to an Auctioneer instance.
  feeWithdrawalDestination?: PublicKey;
  treasuryWithdrawalDestinationOwner?: PublicKey;
  auctioneerAuthority?: PublicKey;
```